src/actions.js

```diff
@@ -121,65 +127,116 @@ export const clearInitWallet = () => ({ type: types.SET_INIT_WALLET, payload: nu
-          || e instanceof hathorLib.errors.MaximumNumberInputsError) {
-        return { success: false, message: e.message };
-      }
-      throw e;
-    }
-  } else {
-    return { success: false, message: ret.message };
-  }
+export const sendTx = (wallet, amount, address, token) => () => {
+  return wallet.sendTransaction(address, amount, token);
 };
 
-export const loadHistory = () => (dispatch) => {
+export const startWallet = (words, pin) => (dispatch) => {
+  // If we've lost redux data, we could not properly stop the wallet object
+  // then we don't know if we've cleaned up the wallet data in the storage
+  walletUtil.cleanLoadedData();
+
+  const connection = new Connection({
+    network: 'mainnet', // app currently connects only to mainnet
+    servers: ['https://mobile.wallet.hathor.network/v1a/'],
+  });
+
+  const beforeReloadCallback = () => {
+    dispatch(activateFetchHistory());
+  }
+
+  const walletConfig = {
+    seed: words,
+    store: STORE,
+    connection,
+    password: pin,
+    pinCode: pin,
+    beforeReloadCallback
+  }
+
+  const wallet = new HathorWallet(walletConfig);
+
+  dispatch(setWallet(wallet));
+
   dispatch(fetchHistoryBegin());
-  hathorLib.version.checkApiVersion().then((data) => {
-    // Save server info.
-    dispatch(setServerInfo({
-      version: data.version,
-      network: data.network,
-    }));
-
-    // Load address history.
-    hathorLib.wallet.loadAddressHistory(0, hathorLib.constants.GAP_LIMIT).then(() => {
-      const walletData = hathorLib.wallet.getWalletData();
-      // Update historyTransactions with new one
-      const historyTransactions = walletData.historyTransactions || {};
-      const { keys } = hathorLib.wallet.getWalletData();
-      dispatch(fetchHistorySuccess(historyTransactions, keys));
-    }, () => {
-      dispatch(fetchHistoryError());
+
+  wallet.start().then((serverInfo) => {
+    dispatch(setServerInfo(serverInfo));
+    wallet.on('state', (state) => {
+      if (state === 4) {
+        // ERROR
+        dispatch(fetchHistoryError());
+      } else if (state === 3) {
+        // READY
+        const historyTransactions = wallet.getTxHistory();
+        const addresses = wallet.getAllAddresses()
+        dispatch(fetchHistorySuccess(historyTransactions, addresses));
+      } else if (state === 0) {
+        // CLOSED
+        // XXX Should we remove the conn event listeners?
+        // we are already closing the connection, is it enough?
+      }
+    })
+
+    wallet.on('new-tx', (tx) => {
+      const addresses = wallet.getAllAddresses()
+      dispatch(newTx(tx, addresses));
+    });
+
+    wallet.on('update-tx', (tx) => {
+      const addresses = wallet.getAllAddresses()
+      dispatch(newTx(tx, addresses));
+    });
+
+    connection.on('height-updated', (height) => {
+      dispatch(updateHeight(height));
     });
-  }, () => {
-    dispatch(fetchHistoryError());
+
+    connection.on('is-online', (value) => {
+      dispatch(setIsOnline(value));
```

Comment:
Can't we do it in the `state` event of the wallet?

Reply:
Yes, we can. Changed it